src/npctalk.cpp

```diff
@@ -185,6 +185,7 @@ struct item_search_data {
     std::vector<str_or_var> material;
     std::vector<str_or_var> flags;
     std::vector<str_or_var> excluded_flags;
+    std::vector<bool> uses_energy;
```

Comment:
there is no need to make it vector, just make it a bool

Reply:
then the documentation need to be updated to reflect it - it was not very obvious for me

Reply:
swapped to optional